examples/test_util/data/EclIOdata.hpp

```diff
@@ -0,0 +1,51 @@
+/*
+  Copyright 2019 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef OPM_ECLIO_DATA_HPP
+#define OPM_ECLIO_DATA_HPP
+
+
+namespace examples {
+
+ namespace test_util {
+
+  namespace data {
+
+    const int sizeOfInte =  4;    // number of bytes pr integer (inte) element
+    const int sizeOfReal =  4;    // number of bytes pr float (real) element
+    const int sizeOfDoub =  8;    // number of bytes pr double (doub) element
+    const int sizeOfLogi =  4;    // number of bytes pr bool (logi) element
+    const int sizeOfChar =  8;    // number of bytes pr string (char) element
+
+    const int MaxNumInte = 1000;    // maximum number of integer elements in binary data block, 4000 bytes, 4 bytes pr element  
+    const int MaxNumReal = 1000;    // maximum number of float elements in binary data block, 4000 bytes, 4 bytes pr element  
+    const int MaxNumDoub = 1000;    // maximum number of double elements in binary data block, 8000 bytes, 8 bytes pr element  
+    const int MaxNumLogi = 1000;    // maximum number of bool elements in binary data block, 4000 bytes, 4 bytes pr element  
+    const int MaxNumChar =  105;    // maximum number of string elements (char[8]) in binary data block, 840 bytes, 8 bytes pr element  
```

Comment:
The way this is organized the `MaxNumxxx`, `sizeOfxxx` and enum values are treated on an equal independent footing; i.e. that they really behave as triplets like: `(INTE, sizeOfInte, MaxNumInte)` is not enforced in the code in any way - you just have to infer it from the variable names. I would recommend either making a small struct/class or small functions like this:
```
int block_size(enum eclArrType arrayType) {
    if (arrayType == INTE)
        return ...
```  

The current implementation works, but when you try to implement the basic reading functionality in one function I think you will see that splitting this information in independent variables in this way is unfortunate.

Reply:
Wil this be OK ? 

New function in EclIOdata

    std::tuple<const int, const int> block_size_data(eclArrType arrType){

       switch(arrType) {
           case INTE  : return std::make_tuple(sizeOfInte,MaxNumInte);   break;
           case REAL  : return std::make_tuple(sizeOfReal,MaxNumReal);   break;
           case DOUB  : return std::make_tuple(sizeOfDoub,MaxNumDoub);   break;
           case LOGI  : return std::make_tuple(sizeOfLogi,MaxNumLogi);   break;
           case CHAR  : return std::make_tuple(sizeOfChar,MaxNumChar);   break;
           case MESS  : throw std::invalid_argument("Type 'MESS' have not assosiated data") ;   break;
       }
    }

which can be called like this 

    auto size = Opm::ecl::block_size_data(Opm::ecl::DOUB);
    
    int sizeOfElement = std::get<0>(size);
    int MaxBlockSize = std::get<1>(size);

    std::cout << "max block size is: " <<   MaxBlockSize << std::endl;  
    std::cout << "size of each element: " <<   sizeOfElement << std::endl;  

I'll implement if you can approve that this is OK.

Reply:
block_size_data has been implemented as a private member function in EclIO

Reply:
> Wil this be OK ?

Something along those lines yes; although I would prefer that you either make two functions returning an int - or alternatively a small class with all the information (type, block_size and element_size) assembled, the current approach is a bit "in the middle", but if you prefer the current approach that is OK;  would change `std::tuple` &rightarrow; `std::pair` though.